### ISSUE TITLE
Merge OSX improvements

### ISSUE THREAD

**Issue by brotzeit** [emacs-ng/emacs-ng#99]
I don't use it myself, but I know there are some features and improvements for mac that won't be merged upstream for whatever reasons. Pull requests are welcome.

**Comment by SebastienWae** [comment]
I'm currently using `emacs-mac` by mituharu.

The biggest improvement is the application-side double buffering with metal, this basically eliminate any flicker, but it might not be needed once `webrender` is merged.

Besides that it's also worth mentioning:
- Image I/O support, will display images inside emacs instead of imagemagick (but it's still needed for `dired-image` anyway)
- AppleScript support
- Touchpad gesture support (can't verify)
- Better clipboard/dnd support (?)
- Better support for the right ⌘ and ⌥ keys (?)

Since emacs flickers too much for me at the moment, I'm waiting on the `webrender`  branch to be fully merged before I can properly test the differences.

The full changelog is here: https://bitbucket.org/mituharu/emacs-mac/src/master/NEWS-mac

**Comment by DavidDeSimone** [comment]
I'm for merging those improvements even with WebRender, just to give users the option in case they pick the --without-webrender configuration.

**Comment by brotzeit** [comment]
> I'm waiting on the webrender branch to be fully merged before I can properly test the differences.

I expect that webrender will take a long time before it's ready for daily use.

**Comment by SebastienWae** [comment]
Then merging `emacs-mac` metal rendering would be great since it would make `emacs-ng` a lot more usable on macOS.

The rest of the features are, imo, nice to have but not a priority.

An other list of features implemted in `emacs-mac`: https://bitbucket.org/mituharu/emacs-mac/src/master/README-mac

**Comment by shaunsingh** [comment]
It would be nice to have a homebrew formula or cask as well, considering how popular homebrew is for macOS developers. I understand its available on nix, but most macOS machines (especially those provided by others) don't have nix available or installable

**Comment by maikol-solis** [comment]
> It would be nice to have a homebrew formula or cask as well, considering how popular homebrew is for macOS developers. I understand its available on nix, but most macOS machines (especially those provided by others) don't have nix available or installable

A hombrew formula/cask would be awesome.
